include/simgrid/s4u/actor.hpp

```diff
@@ -39,10 +42,29 @@ namespace s4u {
  */
 XBT_PUBLIC_CLASS Actor {
   friend Comm;
+  friend int ::s4u_actor_runner(int argc, char **argv);
+
   Actor(smx_process_t smx_proc);
 public:
-  Actor(const char*name, s4u::Host *host, int argc, char **argv);
-  Actor(const char*name, s4u::Host *host, int argc, char **argv, double killTime);
+  
+  /** Method to create a new actor, the template parameter must extend the class Actor. In addition  the inheritance must be public.  */
+  template<class ActorSubClass> static ActorSubClass *createActor(const char*name, s4u::Host *host, int argc, char **argv) {
+    return createActor<ActorSubClass>( name, host, argc, argv, -1);
+  };
+  
+  /** Method to create a new actor, the template parameter must extend the class Actor. In addition  the inheritance must be public. */
+  template<class ActorSubClass> static ActorSubClass *createActor(const char*name, s4u::Host *host, int argc, char **argv, double killTime) {
+     Actor *res = dynamic_cast<Actor*>(new ActorSubClass());
```

Comment:
We can just use `ActorSubClass*` and avoid all the casts here.

We can use a static assert such as:

    static_assert(std::is_base_of<Actor, ActorSubClass>::value, "Actor::createActor only works with subclasses of Actor");

in order to enforce the types.